### ISSUE TITLE
SES subscription (inbound) confirmation error: "Invalid parameter: TopicArn" with non-default AWS region

### ISSUE THREAD

**Issue by mark-mishyn** [anymail/django-anymail#235]
Hello guys!

I'm trying to configure Inbound webhook for AWS.
I follow these steps: [Inbound webhook](https://anymail.readthedocs.io/en/stable/esps/amazon_ses/#inbound-webhook)

I've faced an error on step 3 -- SNS subscription confirmation. When AWS makes a HTTP request to confirm subscription I see this error in my terminal: 
**botocore.errorfactory.InvalidParameterException: An error occurred (InvalidParameter) when calling the ConfirmSubscription operation: Invalid parameter: TopicArn**

To configure test subscription I use Django's runserver and ngrok to make my local app publicly available.

I've tried: update boto3 and botocore to latest versions; re-create topic and subscription; check github issues in this repo, but without success.
 
* **Anymail version:** 8.2
* **ESP:** AWS SES
* **ANYMAIL settings**
```
ANYMAIL = {
    'WEBHOOK_SECRET': 'redacted',
    'AMAZON_SES_CONFIGURATION_SET_NAME': 'TrackingConfigSet',
    'AMAZON_SES_CLIENT_PARAMS': {
        'aws_access_key_id': 'redacted',
        'aws_secret_access_key': 'redacted',
        'region_name': 'us-west-2',  # Oregon
     },
}
```
* **other versions:** 
requests==2.25.1
Django==3.1
python==3.9.1
* **Stack trace**
```
Traceback (most recent call last):
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/django/core/handlers/base.py", line 179, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/django/views/generic/base.py", line 73, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/django/utils/decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/anymail/webhooks/base.py", line 67, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/django/views/generic/base.py", line 101, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/anymail/webhooks/amazon_ses.py", line 83, in post
    return super().post(request, *args, **kwargs)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/anymail/webhooks/base.py", line 81, in post
    events = self.parse_events(request)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/anymail/webhooks/amazon_ses.py", line 102, in parse_events
    self.auto_confirm_sns_subscription(sns_message)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/anymail/webhooks/amazon_ses.py", line 134, in auto_confirm_sns_subscription
    sns_client.confirm_subscription(
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/m/.pyenv/versions/kinder/lib/python3.9/site-packages/botocore/client.py", line 676, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.InvalidParameterException: An error occurred (InvalidParameter) when calling the ConfirmSubscription operation: Invalid parameter: TopicArn
```

**Comment by medmunds** [comment]
Did you create your SNS topic in the same AWS region you've told Anymail to use? ("us-west-2" per your settings.)

Searching the web for "ConfirmSubscription operation: Invalid parameter: TopicArn", it seems like region mismatch is a common way to get this error.

**Comment by mark-mishyn** [comment]
Aha, you are right! My mistake.

We use California region wherever possible in our project, but since only Virginia, Oregon and Ireland SES regions were available in 2018, our SES lives in Oregon.

Thanks a lot!

**Comment by medmunds** [comment]
The more I think about this, there are some valid use cases for receiving SES notifications (inbound or tracking) from multiple regions, or from regions that don't match the boto3 default or explicit Anymail configuration. 

So I'm going to reopen this as an Anymail bug: Anymail should always confirm SNS subscriptions in whatever region the SNS Topic lives.

Note that ["The Amazon SNS topic you choose must be in the same AWS region as the Amazon SES endpoint you use to receive email."](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/receiving-email-action-s3.html) (So your SNS topic would still need to be in Oregon alongside your SES. I *think* AWS would have complained when you tried to attach it as an SES receipt action, step 5 in the Anymail docs.)

**Comment by medmunds** [comment]
Fix released in v8.3
